packages/react-vapor/src/components/slider/MiddleSlider.tsx

```diff
@@ -0,0 +1,97 @@
+import 'rc-slider/assets/index.css';
+
+import {createSliderWithTooltip, Handle, Range, SliderProps} from 'rc-slider';
+import * as React from 'react';
+import {connect} from 'react-redux';
+
+import {IDispatch} from '../../utils/ReduxUtils';
+import {SliderActions} from './SliderActions';
+// import {Tooltip} from '../tooltip/Tooltip';
+
+export interface MiddleSliderOwnProps extends SliderProps {
+    id: string;
+    range: number;
+    defaultValue?: number;
+    hasTooltip?: boolean;
+    tabIndex?: number[];
+}
+
+export const mapDispatchToProps = (dispatch: IDispatch, ownProps: MiddleSliderOwnProps) => ({
+    setOutputValue: (value: number) => dispatch(SliderActions.setValue(ownProps.id, value)),
+});
+
+export const MiddleSliderDisconnected: React.FunctionComponent<MiddleSliderOwnProps &
+    ReturnType<typeof mapDispatchToProps>> = ({range, setOutputValue, hasTooltip}) => {
+    const [highRange, setHighRange] = React.useState(50);
+    const [lowRange, setLowRange] = React.useState(50);
+
+    const valuesAreLowerthanFifty = (value: number[]) => value.every((singleValue) => singleValue < 50);
+    const valuesAreHigherthanFifty = (value: number[]) => value.every((singleValue) => singleValue > 50);
+    const noValuesAreAtFifty = (value: number[]) => value.every((singleValue) => singleValue !== 50);
+    const handlesAreNotInTheCenter = (handleIndex: number) =>
+        (handleIndex === 0 && lowRange <= 50 && highRange === 50) ||
+        (handleIndex === 1 && highRange >= 50 && lowRange === 50);
+
+    React.useEffect(() => {
+        const rangeOutputValue = getComputedRangeValue();
+        setOutputValue(rangeOutputValue);
+    });
+
+    const setRange = (value: number[]) => {
+        if (valuesAreLowerthanFifty(value)) {
+            setValueInTheLowRange(value);
+        } else if (valuesAreHigherthanFifty(value)) {
+            setValueInTheHighRange(value);
+        } else if (noValuesAreAtFifty(value)) {
+            lowRange === 50 ? jumpValueFromHighToLowRange(value) : jumpValueFromLowToHighRange(value);
+        } else {
+            setLowRange(value[0]);
+            setHighRange(value[1]);
+        }
+    };
+
+    const getComputedRangeValue = () => {
+        return lowRange === 50 ? ((highRange - 50) / 50) * range : -Math.abs(((lowRange - 50) / 50) * range);
+    };
+
+    const jumpValueFromHighToLowRange = (value: number[]) => {
+        setHighRange(50);
+        setLowRange(value[0]);
+    };
+
+    const jumpValueFromLowToHighRange = (value: number[]) => {
+        setLowRange(50);
+        setHighRange(value[1]);
+    };
+
+    const setValueInTheLowRange = (value: number[]) => {
+        const correctLowRangeValue = value[1] < 50 ? value[1] : value[0];
```

Comment:
Would love to... 
but look. The ternary checks of the value index 1 is lower than 50. If the already chosen value is... lets say 10. When another value chosen is under the initial value (50) but bugger than 10. If we do not check this and we click nearer to the middle than the lowest handle(like 45), the initial range slider will want to put the handle to the right (index 1) on the wanted position. but since this handle value is forced on the original position (50). The handles will not move.